libgeos/libgeos_handle.go

```diff
@@ -0,0 +1,236 @@
+package libgeos
+
+/*
+#cgo linux CFLAGS: -I/usr/include
+#cgo linux LDFLAGS: -L/usr/lib -lgeos_c
+#include "geos_c.h"
+#include <stdlib.h>
+#include <string.h>
+
+void sf_error_handler(const char *message, void *userdata) {
+	strncpy(userdata, message, 1024);
```

Comment:
Note, `strncpy` writes 1024 bytes to `userdata`, and if the `message` is not '\0' terminated within its first 1024 bytes, `strncpy` will not write a terminating '\0' byte. You may wish to ensure the final byte is a '\0' byte by setting `userdata[1023] = '\0'` yourself manually, depending on whether it is required that it be a '\0' terminated string.

Reply:
Yep, that makes sense.

This should be ok, since we don't require that `userdata` is a null terminated string. It's a pointer to a `[1024]byte` chunk of Go memory, and we don't treat it as necessarily being null terminated when we extract a Go string out of it.